### PR TITLE
Fix comments, add unit tests

### PR DESCRIPTION
**Pull request by mlburggr**
The comments were a little wrong, fixed them

### CODE REVIEW

File: analysis.py [hunk 1]
```diff
@@ -110,9 +110,23 @@ def gpggaParse(self, gpgga, line):
...
+                            "line: " + str(line))
 
+    def verifyChecksum(self, line, checksum):
+        wholeString = ",".join(line)
+        calculatedSum = 0
+        for char in wholeString[1:-5]:
+            calculatedSum = calculatedSum ^ ord(char)
+        checksum = checksum.split("*",1)[1]
+        checksum = checksum[:-2]
+        return int(checksum, 16) == calculatedSum
```
Comment (gjsmo) [inline analysis.py]: Clarify - not particularly obvious that int() converts the hex characters to a numeric type

File: analysis.py [hunk 2]
```diff
@@ -110,9 +110,23 @@ def gpggaParse(self, gpgga, line):
...
         gpgga[8] = line[11] + line[12]
-        gpgga[9] = line[13]
-        print(gpgga)
-        return gpgga
+        gpgga[9] = line[14]
+        
+        if self.verifyChecksum(line, gpgga[9]):
+            return gpgga
+        else:
+            raise Exception("Checksum is incorrect! \n" +
```
Comment (gjsmo) [inline analysis.py]: In the exception, return "Checksum is XX, should be YY" if possible

File: analysis.py (hunk 1 above)
Comment (gjsmo) [inline analysis.py]: Clarify - not particularly obvious that int() converts the hex characters to a numeric type

File: analysis.py [hunk 3]
```diff
@@ -110,9 +110,23 @@ def gpggaParse(self, gpgga, line):
...
+                            "gpgga: " + str(gpgga) + "\n" +
+                            "line: " + str(line))
 
+    def verifyChecksum(self, line, checksum):
+        wholeString = ",".join(line)
+        calculatedSum = 0
+        for char in wholeString[1:-5]:
```
Comment (gjsmo) [inline analysis.py]: Does this just prevent inclusion of the checksum? If so, just don't include the checksum in the concatenated string in line 123.

File: analysis.py (hunk 2 above)
Comment (mlburggr) [reply analysis.py]: Done -- moved the exception raising to the verifyChecksum function. Code is cleaner this way anyways.

File: analysis.py (hunk 1 above)
Comment (mlburggr) [reply analysis.py]: Added comment.

File: analysis.py (hunk 3 above)
Comment (mlburggr) [reply analysis.py]: The checksum is an XOR of all chars between $ and * in the sentence, so this takes everything but the $ and removes the * and everything after it.
Moved the [1:-5] to line 123 and commented it for clarity.

**Review by gjsmo — APPROVED** [review]

**Pull request merged**